File: src/leet-code/neetcode-roadmap/1-arrays-and-hashing/LongestConsecutiveSequence128.java
import java.util.*;

void main() {

}

public int longestConsecutive(int[] nums) {
    if (nums.length == 0) {
        return 0;
    }
    TreeSet<Integer> treeSet = new TreeSet<>();
    for (int num : nums) {
        treeSet.add(num);
    }

    int max = 0;
    int curCount = 0;
    for (int num : treeSet) {
        if (treeSet.contains(num + 1)) {
            curCount++;
            max = Math.max(curCount, max);
        } else {
            curCount = 0;
        }
    }

    return max + 1;
}

File: src/leet_code/neetcode-roadmap/1-arrays-and-hashing/LongestConsecutiveSequence128.kt
import java.util.*

fun longestConsecutive(nums: IntArray): Int {
    if (nums.isEmpty()) {
        return 0
    }
    val treeSet = TreeSet<Int>()
    treeSet.addAll(nums.toList())

    var max = 0
    var curCount = 0
    for (num in treeSet) {
        if (treeSet.contains(num + 1)) {
            curCount++;
            max = maxOf(curCount, max);
        } else {
            curCount = 0;
        }
    }

    return max + 1

}